api/api/controllers/agents_controller.py

```diff
@@ -464,6 +465,69 @@ async def put_upgrade_custom_agent(request, agent_id, pretty=False, wait_for_com
     return web.json_response(data=data, status=200, dumps=prettify if pretty else dumps)
 
 
+async def put_upgrade_multiple_agents(request, list_agents, pretty=False, wait_for_complete=False, wpk_repo=None,
+                                      version=None,
+                                      use_http=False, force=False):
+    """Upgrade agents using a WPK file from online repository.
+
+    :param pretty: Show results in human-readable format
+    :param wait_for_complete: Disable timeout response
+    :param list_agents: List of agent IDs. All posible values since 000 onwards.
+    :param wpk_repo: WPK repository.
+    :param version: Wazuh version to upgrade to.
+    :param use_http: Use protocol http. If it's false use https. By default the value is set to false.
+    :param force: Force upgrade.
+    :return: ApiResponse
+    """
+    f_kwargs = {'list_agents': list_agents,
+                'wpk_repo': wpk_repo,
+                'version': version,
+                'use_http': use_http,
+                'force': force}
+
+    dapi = DistributedAPI(f=agent.upgrade_multiple_agents,
```

Comment:
Change upgrade_multiple_agents for upgrade_agents (new function in framework side)